pennylane/circuit_graph.py

```diff
@@ -439,84 +403,74 @@ def update_node(self, old, new):
         Raises:
             ValueError: if the new :class:`~.Operator` does not act on the same wires as the old one
         """
-        # NOTE Does not alter the graph edges in any way. variable_deps is not changed, _grid is not changed. Dangerous!
+        # NOTE Does not alter the graph edges in any way. variable_deps is not changed, Dangerous!
         if new.wires != old.wires:
             raise ValueError("The new Operator must act on the same wires as the old one.")
 
-        new.queue_idx = old.queue_idx
-        self._graph[self._indices[id(old)]] = new
-        index = self._indices.pop(id(old))
-        self._indices[id(new)] = index
+        self._inds_for_objs[WrappedObj(new)] = self._inds_for_objs.pop(WrappedObj(old))
```

Comment:
Potentially silly thought, but using `WrappedObj(op)` is effectively the same as using `id(op)`, since both are strictly dependent on the memory ref of the op. Why not just continue using `id(op)`? I guess that the circuit would have to be extraordinarily deep for there to be a meaningful difference between the memory used by a dict that uses `WrappedObj` as keys and one that uses `id(op)` as keys.

I guess `WrappedObj` has the benefit of storing a reference to the operator, but I don't think I saw anywhere that was used (correct me if I'm wrong).